lib/ansible/parsing/yaml/objects.py

```diff
@@ -68,6 +68,26 @@ class AnsibleSequence(AnsibleBaseYAMLObject, list):
     pass
 
 
+class AnsibleOctal(AnsibleBaseYAMLObject, int):
+    ''' sub class for octals '''
+
+    def __str__(self):
+        return to_text(oct(self).replace('o', ''))
+
+    def __repr__(self):
+        return repr(str(self))
+
+
+class AnsibleHex(AnsibleBaseYAMLObject, int):
+    ''' sub class for octals '''
```

Comment:
Docstring should be hex, not octal.

Docstrings shold also be more informative of why we need this.

Reply:
The same comment about informative applies to Octal.